### PR TITLE
Matchsettings service and functionality

### LINKED ISSUE #94: MatchSettings service
We need a service that handles match settings, the ability to edit, create and delete match settings. It needs to provide extensive functionality to cover many areas as this will be an important part of the core. This includes game mode options and map list. We also need a way to change the current match settings and load other ones. In addition, having the ability to edit match settings live would be a good idea too.

- [x] Create match settings
- [x] Edit match settings
- [x] Delete match settings
- [x] Match settings builder
- [x] Match settings parser
- [x] Edit match settings live
- [x] Live match settings loading

### PR DESCRIPTION
**Pull request by snixtho**
closes #94 

Provides a complete solution for creating, editing, loading and deleting match settings. This also provides a fluent builder for match settings, which can serialize into XML. An XML parser is also provided, and we can add XML schema validation later on if we want to. In addition to this, the service supports editing match settings live so that you don't have to load files all the time. It can also change between game modes live without a MatchSettings file. And a match manager module has been provided which implements several commands to load and manage match settings, as well as general match flow control.

### CODE REVIEW

File: src/Modules/MatchManagerModule/Controllers/FlowControlCommands.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    [ChatCommand("restartmatch", "Restart the current match.", FlowControlPermissions.RestartMatch)]
+    [CommandAlias("/resmatch", hide: true)]
+    public async Task RestartMatchAsync()
+    {
+        await _flowControl.RestartMatchAsync();
+        await _server.InfoMessageAsync($"{Context.Player.NickName} restarted the match.");
+    }
+
+    [ChatCommand("endround", "Force fnd the current round.", FlowControlPermissions.EndRound)]
```
Comment (Chris92de) [inline src/Modules/MatchManagerModule/Controllers/FlowControlCommands.cs:31]: Typo, should be `Force end`

Comment (AtomicLiquid) [reply src/Modules/MatchManagerModule/Controllers/FlowControlCommands.cs:31]:
```suggestion
    [ChatCommand("endround", "Force end the current round.", FlowControlPermissions.EndRound)]
```

File: src/EvoSC.Common/Interfaces/Services/IMatchSettingsService.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
+﻿using EvoSC.Common.Interfaces.Util;
```
Comment (AtomicLiquid) [inline src/EvoSC.Common/Interfaces/Services/IMatchSettingsService.cs:1]: A suggestion to all these comments is to also add the "exceptions" tag. This gives the implementer of this interface a good idea of what exceptions they have to handle based on the comments provided by us.

File: src/EvoSC.Common/Util/MatchSettings/Builders/MatchSettingsBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+    /// <returns></returns>
+    public MatchSettingsBuilder WithMaps(IEnumerable<IMap> maps)
+    {
+        _maps = maps.ToList();
+        return this;
+    }
+
+    /// <summary>
+    /// Set the settings for this match settings. Make sure to
+    /// call WithMode() first!
```
Comment (AtomicLiquid) [inline src/EvoSC.Common/Util/MatchSettings/Builders/MatchSettingsBuilder.cs:176]: `Requires WithMode() to have been called.`?

File: tests/EvoSC.Common.Tests/Util/MatchSettings/MatchSettingsXmlParserTests.cs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+using Xunit;
+
+namespace EvoSC.Common.Tests.Util.MatchSettings;
+
+public class MatchSettingsXmlParserTests
+{
+    [Fact]
+    public async Task Parses_MatchSettings_With_All_Sections()
+    {
+        var xml = """
```
Comment (AtomicLiquid) [inline tests/EvoSC.Common.Tests/Util/MatchSettings/MatchSettingsXmlParserTests.cs:14]: A potential solution to having the XMLs as strings within the test is to add a "resources" folder under EvoSC.Common.Tests which contains all XML's used for testing purposes.  One of the benefits by doing it this way, is that it's easy for new contributors to locate all XMLs and JSONs that are used within the tests, which again can be used to get a better understanding of what the different XML/JSONs need to look like in order to incorporate them into their code.

Comment (AtomicLiquid) [reply tests/EvoSC.Common.Tests/Util/MatchSettings/MatchSettingsXmlParserTests.cs:14]: In Skatteetaten we would also give the XMLs names similar to their purpose, e.g. `matchsettings_with_all_sections.xml` `matchsettings_missing_attributes.xml`

**Review by AtomicLiquid — APPROVED** [review]
Epic, good work!

**Pull request merged**